test/perf_counters_gtest.cc

```diff
@@ -257,10 +261,16 @@ TEST(PerfCountersTest, MultiThreaded) {
       static_cast<double>(after[0] - before[0]),
       static_cast<double>(after[1] - before[1])};
 
-  // Some extra work will happen on the main thread - like joining the threads
-  // - so the ratio won't be quite 2.0, but very close.
-  EXPECT_GE(Elapsed4Threads[0], 1.9 * Elapsed2Threads[0]);
-  EXPECT_GE(Elapsed4Threads[1], 1.9 * Elapsed2Threads[1]);
+  // The following expectations fail (at least on a beefy workstation with lots
+  // of cpus) - it seems that in some circumstances the runtime of 4 threads 
+  // can even be better than with 2.
+  // So instead of expecting 4 threads to be slower, let's just make sure they
+  // do not differ too much in general (one is not more than 10x than the other).
+  EXPECT_THAT(Elapsed4Threads[0] / Elapsed2Threads[0], AllOf(Gt(0.1), Lt(10)));
+  EXPECT_THAT(Elapsed4Threads[1] / Elapsed2Threads[1], AllOf(Gt(0.1), Lt(10)));
+
+  // EXPECT_GE(Elapsed4Threads[0], 1.9 * Elapsed2Threads[0]);
```

Comment:
can we remove the commented code?  it's helpful for this review but i understand the above comment and i think that's sufficient.

Reply:
Yes, I can remove the commented lines.

Sadly my fixes only seem to solve the test issues on my machine, on the CI machine they still fail as this seems even more restricted....